tests/vm_state_test.go

```diff
@@ -0,0 +1,232 @@
+			storageClass, exists := libstorage.GetRWXFileSystemStorageClass()
+			Expect(exists).To(BeTrue(), "No RWX FS storage class found")
+			kv := util.GetCurrentKv(virtClient)
+			kv.Spec.Configuration.VMStateStorageClass = storageClass
+			tests.UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+
+			By("Creating a migratable Fedora VM with UEFI")
+			vmi := libvmi.NewFedora(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithCloudInitNoCloudNetworkData(libnet.CreateDefaultCloudInitNetworkData()),
+				libvmi.WithUefi(false),
+				libvmi.WithResourceMemory("1Gi"),
+			)
+			vmi.Namespace = util.NamespaceTestDefault
+			if withTPM {
+				By("with persistent TPM enabled")
+				vmi.Spec.Domain.Devices.TPM = &v1.TPMDevice{
+					Persistent: pointer.BoolPtr(true),
+				}
+			}
+			if withEFI {
+				By("with persistent EFI enabled")
+				vmi.Spec.Domain.Firmware = &v1.Firmware{
+					Bootloader: &v1.Bootloader{
+						EFI: &v1.EFI{SecureBoot: pointer.BoolPtr(false), Persistent: pointer.BoolPtr(true)},
+					},
+				}
+			}
+			vm := tests.NewRandomVirtualMachine(vmi, false)
+			vm, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+
+			startVM(vm)
+
+			if withTPM {
+				addDataToTPM(vmi)
+			}
+			if withEFI {
+				addDataToEFI(vmi)
+			}
+
+			for _, op := range ops {
+				switch op {
+				case "migrate":
+					migrateVMI(vmi)
+				case "restart":
+					stopVM(vm)
+					startVM(vm)
+				}
+				if withTPM {
+					checkTPM(vmi)
+				}
+				if withEFI {
+					checkEFI(vmi)
+				}
+			}
+
+			By("Stopping and removing the VM")
+			err = virtClient.VirtualMachine(vm.Namespace).Stop(context.Background(), vm.Name, &v1.StopOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			err = virtClient.VirtualMachine(util.NamespaceTestDefault).Delete(context.Background(), vm.Name, &k8smetav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		},
+			Entry("TPM across migration and restart", true, false, "migrate", "restart"),
+			Entry("TPM across restart and migration", true, false, "restart", "migrate"),
+			Entry("EFI across migration and restart", false, true, "migrate", "restart"),
+			Entry("TPM+EFI across migration and restart", true, true, "migrate", "restart"),
+		)
+		It("[Serial]should remove persistent storage PVC if VMI is not owned by a VM", Serial, func() {
+			By("Setting the backend storage class to the default for RWX FS")
+			storageClass, exists := libstorage.GetRWXFileSystemStorageClass()
+			Expect(exists).To(BeTrue(), "No RWX FS storage class found")
+			kv := util.GetCurrentKv(virtClient)
+			kv.Spec.Configuration.VMStateStorageClass = storageClass
+			tests.UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+
+			By("Creating a VMI with persistent TPM enabled")
+			vmi := tests.NewRandomFedoraVMI()
+			vmi.Namespace = util.NamespaceTestDefault
```

Comment:
You don't need to set the namespace, if you use Create, k8s will automatically set it for you

Reply:
Are you sure it is because of that? @alicefr is right, also at line 196 you assign vmi to the return value of `Create()`, that should set the namespace correctly.

It's not a good practice because it can actually break tests in the future, remember the whole PSA root vs nonroot test namespaces madness?

Reply:
My bad, I removed it from the other test... Not sure where my mind was yesterday :sweat_smile: 
Will fix asap